cr8/clients.py

```diff
@@ -317,13 +324,21 @@ def __init__(self, hosts, conn_pool_limit=25):
         }
         self.__session = None
         self.is_cratedb = True
+        self.session_settings = session_settings or {}
 
     @property
     async def _session(self):
         session = self.__session
         if session is None:
             conn = aiohttp.TCPConnector(**self._connector_params)
             self.__session = session = aiohttp.ClientSession(connector=conn)
+            for setting, value in self.session_settings.items():
+                payload = {'stmt': f'set {setting}={value}'}
+                await _exec(
+                    session,
+                    next(self.urls),
+                    dumps(payload, cls=CrateJsonEncoder)
+                )
```

Comment:
`TCPConnector` afaik internally does some pooling, so this will have the opposite problem of the asyncpg variant in that it only ever gets called once per session initialization instead of once per connection.
It's probably necessary to subclass the `TCPConnector` to customize `_create_connection`.

Running the spec mentioned above shows:

```
cr> select stmt, count(*) from sys.jobs_log group by stmt order by 2 desc ;
+----------------------------------+----------+
| stmt                             | count(*) |
+----------------------------------+----------+
| select count(*) from sys.summits |     1000 |
| set timezone=UTC                 |        2 |
| set application_name=my_app      |        2 |
| ANALYZE                          |        1 |
+----------------------------------+----------+
```

Reply:
Why did you mark this as resolved, there are no changes for this?

Reply:
It's resolved, see the results.

Reply:
The settings were set for multiple clients.

Reply:
Do i miss something here ? What would be the expected output for the spec with concurrency 10 ?

Reply:
It should run `set ..` for each connection it establishes.

With `concurrency = 10` I'd expect there to be more than 1 call in the `sys.jobs_log` output.

You can also verify how many connections it made via:

```
cr> select connections from sys.nodes limit 10;
+-----------------------------------------------------------------------------------------------------------+
| connections                                                                                               |
+-----------------------------------------------------------------------------------------------------------+
| {"http": {"open": 1, "total": 12}, "psql": {"open": 0, "total": 0}, "transport": {"open": 0, "total": 0}} |
+-----------------------------------------------------------------------------------------------------------+
```

The numbers should add up